Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -206,6 +207,53 @@ await container.UpsertItemAsync<TItem>(value, new PartitionKey(value.PartitionKe
             TryLogDebugDetails(_logger, () => $"Deleted: {id}");
         }
 
+        /// <inheritdoc/>
+        public ValueTask<bool> ExistsAsync(string id, string partitionKeyValue = null, CancellationToken cancellationToken = default)
+            => ExistsAsync(id, new PartitionKey(partitionKeyValue ?? id), cancellationToken);
+
+        /// <inheritdoc/>
+        public async ValueTask<bool> ExistsAsync(string id, PartitionKey partitionKey, CancellationToken cancellationToken = default)
+        {
+            Container container =
+                await _containerProvider.GetContainerAsync().ConfigureAwait(false);
+
+            if (partitionKey == default)
+            {
+                partitionKey = new PartitionKey(id);
+            }
+
+            try
+            {
+                ItemResponse<TItem> response =
+                    await container.ReadItemAsync<TItem>(id, partitionKey, cancellationToken: cancellationToken)
+                        .ConfigureAwait(false);
+            }
+            catch (CosmosException e) when(e.StatusCode == HttpStatusCode.NotFound)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<bool> ExistsAsync(Expression<Func<TItem, bool>> predicate, CancellationToken cancellationToken = default)
+        {
+            Container container =
+                await _containerProvider.GetContainerAsync().ConfigureAwait(false);
+
+            IQueryable<TItem> query =
+                container.GetItemLinqQueryable<TItem>()
+                    .Where(predicate.Compose(
+                        item => !item.Type.IsDefined() || item.Type == typeof(TItem).Name, Expression.AndAlso));
+
+            TryLogDebugDetails(_logger, () => $"Read: {query}");
+
+            int count = await query.CountAsync(cancellationToken);
+
+            return count > 1;

```

Comment:
```suggestion
            return count > 0;
```

I think you mean either `count >= 1` or just greater than zero?

Reply:
Yeah I did haha!